azure/functions/eventgrid.py

```diff
@@ -36,3 +43,73 @@ def decode(cls, data: meta.Datum, *, trigger_metadata) -> typing.Any:
             data=body.get('data'),
             data_version=body.get('dataVersion'),
         )
+
+
+class EventGridOutConverter(meta.OutConverter, binding="eventGrid"):
+    @classmethod
+    def check_output_type_annotation(cls, pytype: type) -> bool:
+        valid_types = (str, bytes, azf_eventgrid.EventGridEvent,
+                       typing.List[azf_eventgrid.EventGridEvent])
+        return (meta.is_iterable_type_annotation(pytype, str) or meta.
+                is_iterable_type_annotation(pytype,
+                azf_eventgrid.EventGridEvent) or (isinstance(pytype, type)
+                and issubclass(pytype, valid_types)))
+
+    @classmethod
+    def encode(cls, obj: typing.Any, *, expected_type:
+               typing.Optional[type]) -> typing.Optional[Datum]:
+        if isinstance(obj, str):
+            return meta.Datum(type='string', value=obj)
+
+        elif isinstance(obj, bytes):
+            return meta.Datum(type='bytes', value=obj)
+
+        elif isinstance(obj, azf_eventgrid.EventGridEvent):
+            event_time = None
+            if isinstance(obj.event_time, datetime.datetime):
+                # JSON cannot serialize datetime directly
+                event_time = obj.event_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+            return meta.Datum(
+                type='json',
+                value=json.dumps({
+                    'id': obj.id,
+                    'subject': obj.subject,
+                    'dataVersion': obj.data_version,
+                    'eventType': obj.event_type,
+                    'data': obj.get_json(),
+                    'eventTime': event_time if not None else obj.event_time
+                })
+            )
+
+        elif isinstance(obj, collections.abc.Iterable):
+            msgs = []
+            for item in obj:
+                if isinstance(item, str):
+                    msgs.append(item)
+                elif isinstance(item, azf_eventgrid.EventGridEvent):
+                    msgs.append(json.dumps({'id': item.id,
```

Comment:
Can we create a Python Dict here instead of using json.dumps? Since in line 105, we will dump the whole object into datum. If I'm missing something here, please let me know :)

Reply:
Solved. thanks for the help.